codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpMessage.java

```diff
@@ -36,6 +36,15 @@ protected DefaultHttpMessage(final HttpVersion version) {
 
     /**
      * Creates a new instance.
+     * <p>
+     * <b>Warning!</b> Setting {@code validateHeaders} to {@code false} will mean that Netty won't
+     * validate & protect against user-supplied header values that are malicious.
+     * This can leave your server implementation vulnerable to
+     * <a href="https://cwe.mitre.org/data/definitions/113.html">
+     *     CWE-113: Improper Neutralization of CRLF Sequences in HTTP Headers ('HTTP Response Splitting')
+     * </a>.
+     * When disabling this validation, it is the responsibility of the caller to ensure that the values supplied
+     * do not contain a non-url-escaped carriage return (CR) and/or line feed (LF) characters.
      */
     protected DefaultHttpMessage(final HttpVersion version, boolean validateHeaders, boolean singleFieldHeaders) {
```

Comment:
I believe users need a separate control flags for name/value validation at this level too. Having it only at `DefaultHttpMessage` is not enough if they are using `Default[Full]HttpRequest`/`Default[Full]HttpResponse` or `DefaultLastHttpContent`